sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -437,11 +442,26 @@ private[sql] object HadoopFsRelation extends Logging {
       accessTime: Long,
       blockLocations: Array[FakeBlockLocation])
 
+  private[sql] def mergePathFilter(
+      filter1: Option[PathFilter], filter2: Option[PathFilter]): Path => Boolean = {
+    (filter1, filter2) match {
+      case (Some(f1), Some(f2)) =>
+        (path: Path) => f1.accept(path) && f2.accept(path)
+      case (Some(f1), None) =>
+        (path: Path) => f1.accept(path)
+      case (None, Some(f2)) =>
+        (path: Path) => f2.accept(path)
+      case (None, None) =>
+        (path: Path) => true
+    }
```

Comment:
This can be conciser:

```scala
(filter1 ++ filter2).reduceOption { (f1, f2) =>
  (path: Path) => f1.accept(path) && f2.accept(path)
}.getOrElse {
  (path: Path) => true
}
```